.github/workflows/ci-build-next-java.yml

```diff
@@ -1,32 +1,35 @@
 name: CI Build next Java
 
 on:
-  - push
+  push:
+    branches:
+      - main
+  pull_request:
 
 jobs:
-  build:
+  java-17-compatibility:
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
     steps:
       - name: Checkout the repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Set up JDK 17
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           distribution: 'temurin'
           java-version: 17
-      - name: Cache local Maven repository
-        uses: actions/cache@v2
-        with:
-          path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: |
-            ${{ runner.os }}-maven-
+          cache: 'maven'
       - name: Run tests and build with Maven
-        run: mvn --batch-mode --update-snapshots clean package -DtrimStackTrace=false
+        run: |
+          mvn --batch-mode --update-snapshots clean package -DtrimStackTrace=false \
+              -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
       - name: Publish Test Report
         uses: scacap/action-surefire-report@v1
-        if: ${{ always() }}
+        if: ${{ always() && github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]' }}
```

Comment:
`github.event.pull_request.head.repo.full_name == github.repository` in which case this can be false?

Reply:
This is false in case a pull request from a fork is created. The `action-surefire-report` action would fail in this case, so we skip it.